### PR TITLE
Add physics engine benchmark

### PR DESCRIPTION
**Pull request by Jialn**
Add pyGazebo and pyBullet physics engine benchmark

### CODE REVIEW

File: sim_benchmark/turtlebot.world [hunk 1]
```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0" ?>
+<sdf version="1.5">
```
Comment (bayesian) [inline sim_benchmark/turtlebot.world]: perhaps move to SocialRobot/python/social_bot/worlds

File: sim_benchmark/turtlebot_pybullet.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+t0 = time.time()
+
+interval = 100
+fig = None
+while (1):
+    steps += 1
+
+    for i in range(interval):
+        p.setJointMotorControl2(turtle, 0, p.VELOCITY_CONTROL, targetVelocity=random.random() * -4.0,force=100)
+        p.setJointMotorControl2(turtle, 1, p.VELOCITY_CONTROL, targetVelocity=random.random() * -8.0,force=100)
```
Comment (bayesian) [inline sim_benchmark/turtlebot_pybullet.py]: to be fair, should move this too line outside interval loop.. as in gazebo test case, it only control every interval steps.

File: sim_benchmark/robot_df/turtlebot/model.urdf [hunk 3]
```diff
@@ -0,0 +1,750 @@
...
+    - Stacks    : hexagons
+    - 3d Sensor : kinect
+-->
+<robot name="turtlebot" xmlns:controller="http://playerstage.sourceforge.net/gazebo/xmlschema/#controller" xmlns:interface="http://playerstage.sourceforge.net/gazebo/xmlschema/#interface" xmlns:sensor="http://playerstage.sourceforge.net/gazebo/xmlschema/#sensor" xmlns:xacro="http://ros.org/wiki/xacro">
+
+  <link name="base_link">
+    <visual>
+      <geometry>
+        <!-- new mesh -->
+        <mesh filename="/home/ubuntu/Srcs/SocialRobot/simu_benchmark/robot_df/turtlebot/turtlebot/kobuki_description/meshes/main_body.dae"/>
```
Comment (bayesian) [inline sim_benchmark/robot_df/turtlebot/model.urdf]: do not use hardcoded path.

File: sim_benchmark/README.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+Turtlebot pyBullet with camera sensor (image rendered by CPU) :
+14.50 steps/second
+
+
+Turtlebot pyGazebo without camera sensor:
+175 steps/second
+
+Turtlebot pyGazebo with camera sensor:
+98 steps/second
```
Comment (emailweixu) [inline sim_benchmark/README.md]: We need to figure out why there is such big gap between pybullet and pygazebo, otherwise this benchmark is not convincing.

File: sim_benchmark/.gitignore [hunk 5]
```diff
@@ -0,0 +1,106 @@
+.vscode/*
```
Comment (emailweixu) [inline sim_benchmark/.gitignore]: please clean up unrelated stuff

File: sim_benchmark/README.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
+# Sim Benchmark
```
Comment (emailweixu) [inline sim_benchmark/README.md]: Just "benchmark" for the directory name.

File: sim_benchmark/robot_df/turtlebot/model.urdf [hunk 7]
```diff
@@ -0,0 +1,750 @@
+<?xml version="1.0" ?>
+<!-- =================================================================================== -->
+<!-- |    This document was autogenerated by xacro from robots/kobuki_hexagons_kinect.urdf.xacro | -->
```
Comment (emailweixu) [inline sim_benchmark/robot_df/turtlebot/model.urdf]: Please use the complete url for the original file.
Where did you get this file? Is there any copyright/license information we need to include?

Comment (Jialn) [reply sim_benchmark/robot_df/turtlebot/model.urdf]: I'm sorry to put the model file information. 

The file was auto generated by a script [xacro](http://ros.org/wiki/xacro). Original file is kobuki_hexagons_kinect.urdf.xacro, which comes from ros package [turtlebot_description](http://wiki.ros.org/turtlebot_description).  

The file can be found here:
https://github.com/turtlebot/turtlebot/blob/kinetic/turtlebot_description/robots/kobuki_hexagons_kinect.urdf.xacro

I removed some sensors so that it can be compatible with pybullet. I'm not sure what license information should we include here?

File: sim_benchmark/README.md (hunk 6 above)
Comment (Jialn) [reply sim_benchmark/README.md]: OK, I'll change the name.

File: sim_benchmark/.gitignore (hunk 5 above)
Comment (Jialn) [reply sim_benchmark/.gitignore]: OK.

File: sim_benchmark/README.md (hunk 4 above)
Comment (Jialn) [reply sim_benchmark/README.md]: I'm working on this problem.

File: sim_benchmark/robot_df/turtlebot/model.urdf (hunk 3 above)
Comment (Jialn) [reply sim_benchmark/robot_df/turtlebot/model.urdf]: Pybullet can work with relative path, but pygazebo send an error when loading if relative path is used. I'll find a way.

File: sim_benchmark/turtlebot_pybullet.py (hunk 2 above)
Comment (Jialn) [reply sim_benchmark/turtlebot_pybullet.py]: Yes. will be fixed.

File: sim_benchmark/robot_df/turtlebot/model.urdf (hunk 7 above)
Comment (emailweixu) [reply sim_benchmark/robot_df/turtlebot/model.urdf]: You can include the above information in the file and describe what changes you make to the file. Also include the LICENSE file in the original repo: https://github.com/turtlebot/turtlebot/blob/kinetic/LICENSE

File: sim_benchmark/turtlebot.world (hunk 1 above)
Comment (Jialn) [reply sim_benchmark/turtlebot.world]: Done

File: sim_benchmark/turtlebot_pybullet.py (hunk 2 above)
Comment (Jialn) [reply sim_benchmark/turtlebot_pybullet.py]: Use internal substeps now (the 100 loops will be in C++).

File: sim_benchmark/robot_df/turtlebot/model.urdf (hunk 3 above)
Comment (Jialn) [reply sim_benchmark/robot_df/turtlebot/model.urdf]: Fixed.

File: sim_benchmark/README.md (hunk 4 above)
Comment (Jialn) [reply sim_benchmark/README.md]: It seems that gazebo ignore some dynamics for fixed joints/links. Now I removed the fixed upper two layers for less fixed joints. Only the base/battery pack and wheels left. Now performance of pybullet without camera is better than pygazebo now. Case with camera still need checking. Those removed links will be restored when we can find a way that pygazebo and pybullet do the same thing with fixed joint.

File: sim_benchmark/.gitignore (hunk 5 above)
Comment (Jialn) [reply sim_benchmark/.gitignore]: Done.

File: sim_benchmark/README.md (hunk 6 above)
Comment (Jialn) [reply sim_benchmark/README.md]: Done.

File: sim_benchmark/robot_df/turtlebot/model.urdf (hunk 7 above)
Comment (Jialn) [reply sim_benchmark/robot_df/turtlebot/model.urdf]: LICENSE information and changes added.

**Comment by Jialn** [comment]
@emailweixu @bayesian 上面提到的问题都已修复, 请再看一下,如果没有其他问题就可以merge啦

File: benchmark/README.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
+# Sim Benchmark
+
+A simple benchmark for pygazebo and pybullet simulator.
+
```
Comment (emailweixu) [inline benchmark/README.md]: Please note that pygazebo result is for bullet engine. We should also include benchmark of ODE for bullet.

Comment (Jialn) [reply benchmark/README.md]: Done.
